packages/block-editor/src/components/use-block-drop-zone/index.js

```diff
@@ -422,28 +418,17 @@ export default function useBlockDropZone( {
 				} );
 			},
 			[
-				getAllowedBlocks,
 				targetRootClientId,
-				getBlockNamesByClientId,
-				getDraggedBlockClientIds,
-				getBlockType,
-				getBlocks,
-				getBlockListSettings,
 				dropZoneElement,
 				parentBlockClientId,
-				getBlockIndex,
 				registry,
-				showInsertionPoint,
-				isDragging,
-				startDragging,
```

Comment:
Imo, this is much cleaner:
* Selector is closer to the needed spot.
* Unused selectors will be detected.
* Avoids large dependency arrays and harder to read dependencies.